src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp

```diff
@@ -15,14 +15,20 @@ static constexpr size_t min_slm_size = 256;
 namespace kernel_selector {
 
 static std::pair<size_t, size_t> get_input_bf_size(const fully_connected_params& params) {
-    size_t input_f = params.inputs[0].Feature().v;
-    size_t input_batch = params.inputs[0].Batch().v;
+    auto& input = params.inputs[0];
+    size_t input_f = input.Feature().v;
+    size_t input_batch = input.Batch().v;
+
     // 3D input
     if (params.outputs[0].GetLayout() == DataLayout::bfyx) {
-        input_f = params.inputs[0].Y().v;
-        input_batch = params.inputs[0].Batch().v * params.inputs[0].Feature().v;
+        input_f = input.Y().v;
+        input_batch = input.Batch().v * input.Feature().v;
     }
 
+    // In Some model, input_f could be dynamic in input0. It refers to IFM value of weight.
+    if (input.is_dynamic() && input_f == 0 && params.weights.IFM().v != 0)
+        input_f = params.weights.IFM().v;
```

Comment:
Can't we always use this as input_f? 
Seems that now regardless input0 is dynamic or not, it can just use weight IFM.